### PR TITLE
[IOTDB-1562]incorrect exception processing in insertXXX() API

### PR DESCRIPTION
**Pull request by ShuanglinWu**
catch IOTDBException to fix issue_1562

### CODE REVIEW

**Review by wangchao316 — CHANGES_REQUESTED** [review]
Good Job.Thanks for your contribution.
I have a little question, you need add some IT.

**Comment by cigarl** [comment]
In fact, I think the exception here only throws an `IllegalPathException` on `new PartialPath()`
So we just need to update the current handling of the corresponding exception catch.
In my view，perhaps we should use more specific exception types when handling exceptions ,what do you think？

**Comment by ShuanglinWu** [comment]
Thanks for reminding, I will add some IT tonight.
------------------------------------------------------------------
发件人：wangchao316 ***@***.***>
发送时间：2021年8月17日(星期二) 09:43
收件人：apache/iotdb ***@***.***>
抄　送：ShuanglinWu ***@***.***>; Author ***@***.***>
主　题：Re: [apache/iotdb] [IOTDB-1562]incorrect exception processing in insertXXX() API (#3758)

@wangchao316 requested changes on this pull request.
Good Job.Thanks for your contribution.
 I have a little question, you need add some IT.
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by ShuanglinWu** [comment]
I think now it's just IllegalPathException, but maybe there will throws an MisMatchDataTypeException or an PathNOtExistException.
        It seems to me that we don't need to catch each exception, but we should get the error message and error code and return them to the client.
------------------------------------------------------------------
发件人：lisijia ***@***.***>
发送时间：2021年8月17日(星期二) 10:31
收件人：apache/iotdb ***@***.***>
抄　送：ShuanglinWu ***@***.***>; Author ***@***.***>
主　题：Re: [apache/iotdb] [IOTDB-1562]incorrect exception processing in insertXXX() API (#3758)

In fact, I think the exception here only throws an IllegalPathException on new PartialPath()
 So we just need to update the current handling of the corresponding exception catch.
 In my view，perhaps we should use more specific exception types when handling exceptions ,what do you think？
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

File: server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java [hunk 1]
```diff
@@ -2155,6 +2189,19 @@ private TSStatus onNPEOrUnexpectedException(
     return RpcUtils.getStatus(statusCode, message + e.getMessage());
   }
 
+  private TSStatus onIOTDBException(Exception e, String operation, int errorCode) {
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java:2192]:
```suggestion
  private TSStatus onIoTDBException(Exception e, String operation, int errorCode) {
```

File: server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java [hunk 2]
```diff
@@ -1372,6 +1372,11 @@ public TSStatus insertRecords(TSInsertRecordsReq req) {
           allCheckSuccess = false;
         }
         insertRowsPlan.addOneInsertRowPlan(plan, i);
+      } catch (IoTDBException e) {
+        allCheckSuccess = false;
+        insertRowsPlan
+            .getResults()
+            .put(i, onIOTDBException(e, "inserting records", e.getErrorCode()));
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java:1379]: Better to move all these "inserting records" to a IoTDBConstant or a n Enum class

**Review by qiaojialin — APPROVED** [review]

File: session/src/test/java/org/apache/iotdb/session/IoTDBSessionSimpleIT.java [hunk 3]
```diff
@@ -726,6 +725,278 @@ public void testInsertOneDeviceRecordsWithIncorrectOrder()
...
+          tsEncodings,
+          CompressionType.SNAPPY,
+          Arrays.asList("alias1", "alias2", "alias3"));
+      fail("Exception expected");
+    } catch (StatementExecutionException e) {
+      assertTrue(
+          e.getMessage()
+              .contains(
+                  String.format(
+                      "[%s] Exception occurred: %s failed. %s is not a legal path",
```
Comment (qiaojialin) [inline session/src/test/java/org/apache/iotdb/session/IoTDBSessionSimpleIT.java:943]: extract a constant string in this class

**Review by qiaojialin — APPROVED** [review]

**Review by wangchao316 — APPROVED** [review]
LGTM

**Review by HTHou — APPROVED** [review]

**Pull request merged**